### ISSUE TITLE
Client .gz download deadlock

### ISSUE THREAD

**Issue by jackd248** [lochmueller/staticfilecache#225]
## Bug Report

**Current Behavior**
After upgrading from 9 to 10, changing some of the extension settings (sfc_*, Files location, etc.) we got massive problems with browser cache entries which are trying to download gzip files instead of loading the cached websites. The user gets into a deadlock until the browser data/cache is cleared. 

**Environment**
- TYPO3 version: 9.5.11
- staticfilecache version: 10.1.0

**Additional context**
I cannot clearly identify the origin of the problem, but it seems to be related to the upgrade process.

**Comment by jackd248** [comment]
The response header for the bad request:
```
Accept-Ranges: bytes
Cache-Control: max-age=2592000
Content-Encoding: gzip
Content-Length: 19202
Content-Type: application/x-gzip
Date: Tue, 19 Nov 2019 08:37:20 GMT
Expires: Thu, 19 Dec 2019 08:37:20 GMT
Last-Modified: Tue, 19 Nov 2019 07:24:50 GMT
Server: Apache
X-Content-Type-Options: nosniff
X-UA-Compatible: IE=edge
```

**Comment by MartinH0** [comment]
The solution is to modify your htaccess.

Pls insert after:
`# Set proper content type and encoding for gzipped html.`

This snipped to tell the .gzip is a gziped file:

```
<FilesMatch "\.gzip$">
   SetEnv no-gzip 1
   SetEnv no-brotli 1
   <IfModule mod_headers.c>
      Header set Content-Encoding gzip
   </IfModule>
</FilesMatch>
```
Now your Files will be served how they should be.
Alternatively you can disable compression! (But better solve the problem)

**Comment by MartinH0** [comment]
@jackd248 could you pls check out if this helps you?
https://github.com/lochmueller/staticfilecache/issues/227#issuecomment-557816464
This solved a similar issue here on Git yesterday

**Comment by jackd248** [comment]
Currently I'm trying to reproduce the problem to figure out a possible solution.

I updated the htaccess before as mentioned for v10. The problematic files were also served as .gz files and not .gzip. I thought the lines for this `FilesMatch` were already in the htaccess.

**Comment by jackd248** [comment]
**I found the problem**

The issue was not caused by some htaccess changes, but by some special server environments. I moved the SFC directory to another location, where the vhost denied overrides by further htaccess files. I didn't noticed the htaccess files in every subdirectory of the SFC folder, so the important content type `text/html` was missing from every cache entry. 

Thanks for your help, @MartinH0.
